src/KUSignalLib/DSP/PLL.py

```diff
@@ -20,14 +22,33 @@ def __init__(self, kp=1, k0=1, k1=0, k2=0, wstart=1, thetaStart=0, fs=1):
         :param fs: Float type. Sampling frequency.
         Initialize the PLL object for repeated use, if left blank the object will be initialized with default values.
         '''
-        self.Kp = kp
-        self.K0 = k0
-        self.K1 = k1
-        self.K2 = k2
+        if loop_bandwidth == None and damping_factor == None:
+            self.Kp = kp
+            self.K0 = k0
+            self.K1 = k1
+            self.K2 = k2
+        else:
+            self.compute_loop_constants(loop_bandwidth, damping_factor, 1/fs, k0)
         self.w0 = wstart
         self.phase = thetaStart
         self.sigOut = np.exp(1j * thetaStart)
         self.fs = fs
+        
+    def compute_loop_constants(self, loopBandwidth, dampingFactor, T, K0=1):
+        """
+        :param loopBandwidth: Float type. Loop bandwidth.
+        :param dampingFactor: Float type. Damping factor.
+        :param T: Float type. this can be your sampleling peiod(i.e. 1/fs), or in communication systems it
+        can be your symbol time / N (where N is bits sample per symbol) for a higher bandwidth design.
+        Compute the loop filter gains based on the loop bandwidth and damping factor.
+        """
+        theta_n = loopBandwidth*T/(dampingFactor + 1/(4*dampingFactor))
+        K0_Kp = (4*theta_n)/(1+2*dampingFactor*theta_n+theta_n**2)
+        self.K1 = dampingFactor
```

Comment:
i thought that in the discrete domain K1 and K2 needed to be calculated according to....

K1 = ((4 * zeta) / (zeta + (1 / (4 * zeta)))) * ((B * (1 / fs)) / K0)  # Loop filter coefficient 1
K2 = ((4) / (zeta + (1 / (4 * zeta))) ** 2) * (((B * (1 / fs)) ** 2) / K0)  # Loop filter coefficient 2

.... the damping factor is used in the calculations but there is more to it